ui/app/AppLayouts/Wallet/panels/ActivityFilterPanel.qml

```diff
@@ -0,0 +1,264 @@
+                                      return qsTr("Bridge")
+                                  default:
+                                      ""
+                                  }
+            iconAsset.icon: switch(root.typeFilters[index]) {
+                            case Constants.TransactionType.Send:
+                                return "send"
+                            case Constants.TransactionType.Receive:
+                                return "receive"
+                            case Constants.TransactionType.Buy:
+                                return "token"
+                            case Constants.TransactionType.Swap:
+                                return "swap"
+                            case Constants.TransactionType.Bridge:
+                                return "bridge"
+                            default:
+                                ""
+                            }
+            onClosed: root.toggleType(type)
+        }
+    }
+
+    Repeater {
+        model: root.statusFilters.length !== 4 ? root.statusFilters: 0
+        delegate: ActivityFilterTagItem {
+            property int type: root.statusFilters[index]
+            tagPrimaryLabel.text: switch(root.statusFilters[index]) {
+                                  case Constants.TransactionStatus.Failed:
+                                      return qsTr("Failed")
+                                  case Constants.TransactionStatus.Pending:
+                                      return qsTr("Pending")
+                                  case Constants.TransactionStatus.Complete:
+                                      return qsTr("Complete")
+                                  case Constants.TransactionStatus.Finished:
+                                      return qsTr("Finalised")
+                                  default:
+                                      ""
+                                  }
+            iconAsset.icon: switch(root.statusFilters[index]) {
+                            case Constants.TransactionStatus.Failed:
+                                return Style.svg("transaction/failed")
+                            case Constants.TransactionStatus.Pending:
+                                return Style.svg("transaction/pending")
+                            case Constants.TransactionStatus.Complete:
+                                return Style.svg("transaction/verified")
+                            case Constants.TransactionStatus.Finished:
+                                return Style.svg("transaction/finished")
+                            default:
+                                ""
+                            }
+            iconAsset.color: "transparent"
+            onClosed: root.toggleStatus(type)
+        }
+    }
+
+    Repeater {
+        model: root.assetsList
+        delegate: ActivityFilterTagItem {
+            tagPrimaryLabel.text: symbol
+            iconAsset.icon: Style.png("tokens/" + symbol)
```

Comment:
This will yield lots of warnings for tokens which we don't have icons for... Imo we have a function for that somewhere

Reply:
`Constants.tokenIcon(symbol)` yw :)

Reply:
@Cuteivist @Khushboo-dev-cpp ^^